src/js/samples/components/Create/CreateSample.tsx

```diff
@@ -126,50 +129,68 @@ type formValues = {
     sidebar: { labels: number[]; subtractionIds: string[] };
 };
 
-const getInitialValues = forceGroupChoice => ({
-    name: "",
-    isolate: "",
-    host: "",
-    locale: "",
-    libraryType: "normal",
-    readFiles: [],
-    group: forceGroupChoice ? "none" : null,
-    sidebar: { labels: [], subtractionIds: [] },
-});
-
-export function CreateSample({
-    error,
-    forceGroupChoice,
-    groups,
-    subtractions,
-    allLabels,
-    onLoadSubtractionsAndFiles,
-    onCreate,
-    onClearError,
-    onListLabels,
-}) {
-    useEffect(() => {
-        onLoadSubtractionsAndFiles();
-        onListLabels();
-    }, []);
-
-    const { data: readsResponse, isLoading: isReadsLoading }: { data: UnpaginatedFileResponse; isLoading: boolean } =
-        useListFiles(FileType.reads, false);
+function getInitialValues(forceGroupChoice) {
+    return {
+        name: "",
+        isolate: "",
+        host: "",
+        locale: "",
+        libraryType: "normal",
+        readFiles: [],
+        group: forceGroupChoice ? "none" : null,
+        sidebar: { labels: [], subtractionIds: [] },
+    };
+}
 
-    if (subtractions === null || allLabels === null || isReadsLoading) {
+export default function CreateSample() {
+    const { data: allLabels, isLoading: labelsLoading } = useFetchLabels();
+    const { data: groups, isLoading: groupsLoading } = useListGroups();
+    const { data: subtractions, isLoading: subtractionsLoading } = useSubtractionsShortlist();
+    const { data: settings, isLoading: settingsLoading } = useFetchSettings();
+    const {
+        data: readsResponse,
+        isLoading: isReadsLoading,
+        isFetchingNextPage,
+        fetchNextPage,
+    } = useInfiniteFindFiles(FileType.reads, 11);
+
+    const history = useHistory();
+    const samplesMutation = useMutation(createSample, {
+        onSuccess: () => {
+            history.push("/samples");
+        },
+    });
+
+    if (isReadsLoading || labelsLoading || subtractionsLoading || settingsLoading || groupsLoading) {
         return <LoadingPlaceholder margin="36px" />;
     }
 
-    const reads = filter(readsResponse.documents, { reserved: false });
+    function onCreate(name, isolate, host, locale, libraryType, subtractionIds, files, labels, group?) {
+        samplesMutation.mutate({
+            name,
+            isolate,
+            host,
+            locale,
+            libraryType,
+            subtractions: subtractionIds,
+            files,
+            labels,
+            group,
+        });
+    }
 
-    const autofill = (selected, setFieldValue) => {
+    const forceGroupChoice = settings.sample_group === "force_choice";
+
+    const reads = readsResponse.pages.flatMap(page => page.items, { reserved: false });
```

Comment:
```suggestion
    const reads = readsResponse.pages.flatMap(page => page.items, { reserved: false });
```

Is `{ reserved: false }` doing anything here? Also a plug for lodash, all of these default functions, often with conveniences and many more on top. https://lodash.com/docs/4.17.15#flatMap